docs/contracts/abstract-contracts.rst

```diff
@@ -6,12 +6,15 @@
 Abstract Contracts
 ******************
 
-Contracts need to be marked as abstract when at least one of their functions is not implemented.
-Contracts may be marked as abstract even though all functions are implemented.
-
-This can be done by using the ``abstract`` keyword as shown in the following example. Note that this contract needs to be
-defined as abstract, because the function ``utterance()`` was defined, but no implementation was
-provided (no implementation body ``{ }`` was given).
+Contracts must be marked as abstract when at least one of their functions is not implemented or when
+their constructors (or declarations) do not provide arguments for all of their base contract constructors.
+A contract may be marked abstract even if it must not be marked abstract, such as when you do not intend
```

Comment:
I'm used to RFC's etc where the words MAY and MUST are often capitalized and have very specific meanings. Something "may" be a certain way, even when it "must" not be that way. Your suggestion is totally fine, but perhaps it's enough to just changedthe word "marked" to "declared" in my original ... and/or in yours.

From a workflow perspective, I'm not sure if it's up to me to make suggested changes to my PRs and then resubmit them?? Or will you just make the change and accept it??

Reply:
> I'm used to RFC's etc where the words MAY and MUST are often capitalized and have very specific meanings. 

To be honest, the problem of these docs that they're a mix between high-level user guide and a low-level language spec. At least as it is now, I don't really see the strict use as in RFCs fitting there. I think it's good to remove ambiguities, but it should also read well in plain English.

By the way, there's an ongoing initiative to create a more rigorous language spec. See #12739.

> Something "may" be a certain way, even when it "must" not be that way. Your suggestion is totally fine, but perhaps it's enough to just changedthe word "marked" to "declared" in my original ... and/or in yours.

My problem with your version was really the conflict between "may" and "must" here and I'm not sure how the "marked"/"declared" change would help with that. I.e. it was written as if you meant "not(must(be marked abstract))" but I read it as "must(not(be marked abstract))" and I think this is the usual interpretation of "must not" in English.

> From a workflow perspective, I'm not sure if it's up to me to make suggested changes to my PRs and then resubmit them?? Or will you just make the change and accept it??

Not sure I fully understand the question, are you asking about the suggested edits I added in my comments? It's up to you to incorporate them or not. You can just take them directly, rewrite them your way or just ignore them if you think they make no sense. We can discuss them until all the threads here are resolved. Then we can approve and merge the PR. The new changes can be in new commits if that's more convenient to you but the commits will need to be squashed before the PR is merged anyway.

By the way, it would be best to mark these comment threads as resolved only once both the change and the discussion is  fully resolved. I.e. if it's a simple change with no further discussion needed, it's fine to resolve them. If not, as is the case in this one, it's also fine to keep them open so that reviewer can spot them.

Reply:
Oh!  I see the confusion that my text may create now.